google-cloud-examples/src/test/java/com/google/cloud/examples/bigtable/ITHelloWorld.java

```diff
@@ -0,0 +1,130 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.AfterClass;
+import org.junit.AssumptionViolatedException;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ITHelloWorld {
+
+  private static final String INSTANCE_PROPERTY_NAME = "bigtable.instance";
+  private static final String TABLE_PREFIX = "table";
+  private static String tableId;
+  private static BigtableDataClient dataClient;
+  private static BigtableTableAdminClient adminClient;
+  private static InstanceName instanceName;
+  private HelloWorld helloWorld;
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    String targetInstance = System.getProperty(INSTANCE_PROPERTY_NAME);
+    if (targetInstance == null) {
+      dataClient = null;
+      adminClient = null;
+      return;
+    }
+    instanceName = InstanceName.parse(targetInstance);
+    BigtableDataSettings settings =
+        BigtableDataSettings.newBuilder().setInstanceName(instanceName).build();
+    dataClient = BigtableDataClient.create(settings);
+    BigtableTableAdminSettings adminSettings =
+        BigtableTableAdminSettings.newBuilder()
+            .setInstanceName(com.google.bigtable.admin.v2.InstanceName.parse(targetInstance))
+            .build();
+    adminClient = BigtableTableAdminClient.create(adminSettings);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    garbageCollect();
+    adminClient.deleteTable(tableId);
+    dataClient.close();
+    adminClient.close();
+  }
+
+  @Before
+  public void setup() throws IOException {
+    if (adminClient == null || dataClient == null) {
+      throw new AssumptionViolatedException(
+          INSTANCE_PROPERTY_NAME + " property is not set, skipping integration tests.");
+    }
+    tableId = generateTableId();
+    helloWorld = new HelloWorld(instanceName.getProject(), instanceName.getInstance(), tableId);
+    adminClient.createTable(CreateTableRequest.of(tableId).addFamily("cf1"));
+  }
+
+  @Test
+  public void testCreateAndDeleteTable() throws IOException {
+    // Create table
+    String fakeTable = generateTableId();
+    HelloWorld testHelloWorld =
+        new HelloWorld(instanceName.getProject(), instanceName.getInstance(), fakeTable);
+    testHelloWorld.createTable();
+    assertTrue(adminClient.exists(fakeTable));
+
+    // Delete table
+    testHelloWorld.deleteTable();
+    assertTrue(!adminClient.exists(fakeTable));
+  }
+
+  @Test
+  public void testWriteToAndReadFromTable() {
+    // Write to table
+    helloWorld.writeToTable();
+    Row row = dataClient.readRow(tableId, "rowKey0");
+    assertNotNull(row);
+  }
+
+  private String generateTableId() {
+    return String.format(
+        "%s-%016x-%d", TABLE_PREFIX, System.currentTimeMillis(), new Random().nextLong());
+  }
+
+  private static void garbageCollect() {
+    Pattern timestampPattern = Pattern.compile(TABLE_PREFIX + "-([0-9]+)");
```

Comment:
This doesnt match the generateTableId format above. The correct format is:
TABLE_PREFIX + "-([0-9a-f]+)-([0-9a-f]+)"